microprofile-jwt-auth-tool/src/main/java/org/jboss/eap/qe/microprofile/jwt/auth/tool/JoseHeader.java

```diff
@@ -0,0 +1,91 @@
+package org.jboss.eap.qe.microprofile.jwt.auth.tool;
+
+import javax.json.Json;
+import javax.json.JsonObject;
+
+/**
+ * A class representing a header of JWT
+ */
+public class JoseHeader {
+
+    private final String keyId;
+    private final String type;
+    private final String algorithm;
+
+    private JoseHeader(Builder builder) {
+        this.keyId = builder.keyId;
+        this.type = builder.type;
+        this.algorithm = builder.algorithm;
+    }
+
+    /**
+     * Shortcut for creating valid instance with custom key ID. Useful when you just need a header with valid values.
+     * @param keyId ID of key which will be propagated in this header.
+     * @return new instance of valid header
+     */
+    public static JoseHeader validWithKeyId(final String keyId) {
+        return new JoseHeader.Builder()
+                .keyId(keyId)
+                .algorithm("RS256")
+                .type("JWT")
+                .build();
+    }
+
+    /**
+     * Format values to JSON
+     * @return JSON representation
+     */
+    public JsonObject toJson() {
+        return Json.createObjectBuilder()
+                .add("kid", this.keyId)
+                .add("typ", this.type)
+                .add("alg", this.algorithm)
+                .build();
+    }
+
+    public static final class Builder {
```

Comment:
All all of the fields required? If so, there should be some kind of check. Moreover, If we do not expect more fields in the future, simple constructor of three required fields is simpler, shorter and more readable.

Reply:
Thanks - that makes sense. The evolution of this class is dependent on the evolution of specification. Though I can get rid of this builder pattern for sure.

Reply:
Thanks. Right now it is just too complex.

Reply:
Working on it right now.